### ISSUE TITLE
Make "polygon" data type consistent across polygon types

### ISSUE THREAD

**Issue by Black-Virtue** [godotengine/godot-proposals#5536]
### Describe the project you are working on

Trying to draw various non-visual polygons to viewports, outside of editor view.

### Describe the problem or limitation you are having in your project

Currently several nodes and built in classes represent polygons and many of them have a polygon properties (or "add","get" and/or "set" polygon(s) methods)
The data types of these are inconsistent, making conversion awkward.

eg. Polygon2d.polygon is a PackedVector2Array while NavigationPolygon's add_polygon and get_polygon use PackedInt32Array.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Making a unified Polygon class (or one each for 2d/3d) to represent these polygons would avoid conversion issues.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

In short term, adding in a method to convert polygons to/from the varying standards.
Longer term, moving the expected and returned data structures to a consistent format.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

This can be worked around with script, however from a usability and consistency stance does need work.

### Is there a reason why this should be core and not an add-on in the asset library?

This is a core part of Godot's data handling
